ddtrace/appsec/_asm_request_context.py

```diff
@@ -65,31 +51,36 @@ def __init__(self, active=False):  # type: (bool) -> None
         self.addresses_sent = set()  # type: set[str]
 
 
-_ASM = contextvars.ContextVar("ASM_contextvar", default=ASM_Environment())
+def _get_asm_context():
+    env = core.get_item("asm_env")
+    if env is None:
+        env = ASM_Environment()
+        core.set_item("asm_env", env)
+    return env
 
 
 def free_context_available():  # type: () -> bool
-    env = _ASM.get()
+    env = _get_asm_context()
     return env.active and env.span is None
 
 
 def in_context():  # type: () -> bool
-    env = _ASM.get()
+    env = _get_asm_context()
     return env.active
 
 
 def is_blocked():  # type: () -> bool
     try:
-        env = _ASM.get()
+        env = _get_asm_context()
         if not env.active or env.span is None:
             return False
-        return _context.get_item(WAF_CONTEXT_NAMES.BLOCKED, span=env.span)
+        return bool(core.get_item(WAF_CONTEXT_NAMES.BLOCKED, span=env.span))
```

Comment:
Was the method before not always returning a bool and we're fixing that here?

Reply:
Oh, it looks like before we would just set it as a bool, but there was no guarantee anywhere that we had to do that, so we convert it just to be sure.